### ISSUE TITLE
[npm-install] <help no very helpful>

### ISSUE THREAD

**Issue by 1C0D** [npm/cli#6610]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This is a CLI Docs Enhancement, not another kind of Docs Enhancement.

- [X] This is a CLI Docs Enhancement.

### Description of Problem

the help doesn't explain what is happening in the package.json when doing npm install module_name and how to deal with this... how to clean useless entries etc...

### Potential Solution

More complete explanations for beginners

### Docs URL

https://docs.npmjs.com/cli/v8/commands/npm-install

**Comment by ljharb** [comment]
What do you mean "useless entries"?

**Comment by 1C0D** [comment]
I mean the interactions between package.json and npm install. Apparently now no options are needed to modify the package (--save). But if we are doing --save-dev what is happening? why doing it? how to reverse it... I see some repos with some dependencies in devDeps but it seems to be for dev part as install lodash... but it's imported in the code so this is still needed. how devDeps can be used? ... I asked some info to GPT but trying to read helps I didn't find what I needed.

**Comment by 1C0D** [comment]
Well I found this page https://docs.npmjs.com/specifying-dependencies-and-devdependencies-in-a-package-json-file
but it's hard to find. it's one of the most important things to know what we are doing and why and how to modify it. The part about why and when you chose devDeps or deps would be necessary for beginners and apparently more confirmed users. Searching to understand the logic behind, I found many errors on github. and some installations are confusing as npm i --save-dev @types/lodash.
